src/read.jl

```diff
@@ -109,16 +109,33 @@ function read_demand_flexibility(filepath, interval)::DemandFlexibility
     # Initialize demand flexibility
     demand_flexibility = Dict()
 
-    # Try loading demand flexibility profiles. Otherwise, create a DataFrame of zeros
-    try
-        demand_flexibility["flex_amt"] = CSV.File(
-            joinpath(filepath, "demand_flexibility.csv")
-        ) |> DataFrames.DataFrame
-        println("...loading demand flexibility profiles")
+    # Try loading demand flexibility profiles
+    for s in ["up", "dn"]
+        try
+            demand_flexibility["flex_amt_" * s] = CSV.File(
+                joinpath(filepath, "demand_flexibility_" * s * ".csv")
+            ) |> DataFrames.DataFrame
+            println("...loading demand flexibility " * s * " profiles")
+        catch e
+            println("Demand flexibility " * s * " profile not found in " * filepath)
+            demand_flexibility["flex_amt_" * s] = nothing
+        end
+    end
+    
+    # If both demand flexibility profiles exist, demand flexibility is enabled
+    if !isnothing(demand_flexibility["flex_amt_up"]) && (
+        !isnothing(demand_flexibility["flex_amt_dn"])
+    )
         demand_flexibility["enabled"] = true
-    catch e
-        println("Demand flexibility profiles not found in " * filepath)
-        demand_flexibility["flex_amt"] = nothing
+    else
+        if !isnothing(demand_flexibility["flex_amt_up"]) || (
+            !isnothing(demand_flexibility["flex_amt_dn"])
+        )
+            @warn (
+                "The exclusion of one of the demand flexibility profiles has resulted "
+                * "in demand flexibility not being enabled."
+            )
+        end
```

Comment:
If we know that at least one of these is `nothing` from the `if` block above, do we need this `if` block here within the `else`?

Reply:
Oh, I think I see now. This block is triggered if exactly one is present and exactly ones is missing.

Reply:
Yeah, that was the intent! If both are `nothing`, then the user likely doesn't intend for demand flexibility to be present. However, if only one is `nothing`, the user may have forgotten to include one of the demand flexibility profiles. When one profile is excluded, demand flexibility won't work in the way that we have it set up, which is why we set `enabled` to `false` and include a more prominent warning to the user. If you would like, I can get rid of that `if` block and change the warning message to be `"The exclusion of at least one demand flexibility profile ..."`

Reply:
I think the current behavior makes sense, it just took me a minute to parse the code properly in my head.

Reply:
@lanesmith The logic makes sense to me. My questions is in the current setup, is `demand_flexibility` a parameter that the user could specify or it is still an internal indicator? If the user could set `demand_flexibility` parameter optionally, the current behavior could serve as the default behavior when it is not set explicitly by the user, i.e. `isnothing`. Otherwise, following situation still applies: the user would like to have demand flexibility, thus the parameter is set, however, none of the profiles are provided by accident.

Reply:
I think what you are describing is the `enabled` parameter, which belongs to the `DemandFlexibility` struct and is specified in the `demand_flexibility_parameters.csv` file. So, yes, users can specify whether or not they want demand flexibility. However, if they don't have the requisite files (namely, the `demand_flexibility_up.csv` and `demand_flexibility_dn.csv` files), then demand flexibility cannot be enabled, regardless of the user's preference. There are a couple reasons I like checking for the demand flexibility profiles first, as opposed to checking for the `enabled` parameter (and, by extension, the `demand_flexibility_parameters.csv` file) first: (1) `demand_flexibility_profiles.csv` is not necessarily a required file meaning that, at a minimum, a user would only need to provide the `demand_flexibility_up.csv` and `demand_flexibility_dn.csv` files; and (2) if the profiles are not included, trying to read in the other files is not required. I'm happy to discuss this flow more if we would like to rearrange when things are read.

Reply:
What I'm trying to say here is:

If the user specify `demand_flexibility_enabled = True`, but the profiles are not provided, we should throw an error instead of assuming `demand_flexibility_enabled = False` then running the simulation without it.

If the user specify `demand_flexibility_enabled = False`, but the profiles are provided, we ignore it. (I think it is the case in the current form)

If the user doesn't specify the parameter, i.e. `demand_flexibility_enabled == nothing`, the current logic makes perfect sense.

Reply:
Okay, that makes sense. I'll make that change. In the first case that you mention, should it raise an error (causes the simulation to stop) or a warning (prints the warning, but the simulation continues using updated parameters set internally)?

Reply:
I think if the user explicitly set `demand_flexibility_enabled = True` without providing the required profiles, we should raise an error and stop the simulation.

Reply:
This should be done, though please look over `read_demand_flexibility` as it was refactored quite a bit. The way I was able to get the error message to be called is a little weird. When `throw`-ing an error, the resultant error at the top of the stacktrace is that `m` is not defined. In Julia, you can see the desired error text later in the stacktrace by just using `throw(ErrorException(...))`, however in Python you cannot (which kind of defeats the purpose of having the error message); apparently, Python does not produce as detailed of a stacktrace for the errors in the Julia code. So, that is why I have the main error message attached to `@error(...)`, which acts more like an error-centric print statement (it does not stop the simulation), and a minimal error message attached to `throw(ErrorException(...))`, which actually stops the simulation. If anybody knows of a better solution, I'd be happy to implement it